src/main/java/org/terracotta/offheapstore/OffHeapHashMap.java

```diff
@@ -1410,8 +1410,9 @@ private void cleanPendingTables(int hash, long encoding) {
   
           if (isTerminating(entry)) {
             break;
-          } else if (isPresent(entry) && (hash == entry.get(KEY_HASHCODE)) && (encoding == readLong(entry, ENCODING))) {
-            entry.put(STATUS, STATUS_REMOVED);
+          } else if (isPresent(entry) && (hash == entry.get(KEY_HASHCODE)) && (oldEncoding == readLong(entry, ENCODING))) {
+            entry.put(newEntry);
+            newEntry.flip();
```

Comment:
Is this `flip()` necessary or proper?  The calls to `updatePendingTables` from `slotUpdated` all have "flipped" buffers.